src/garage/experiment/experiment_wrapper.py

```diff
@@ -182,24 +193,32 @@ def run_experiment(argv):
     logger.add_output(dowel.CsvOutput(tabular_log_file))
     logger.add_output(dowel.TensorBoardOutput(log_dir))
     logger.add_output(dowel.StdOutput())
-    prev_snapshot_dir = snapshotter.snapshot_dir
-    prev_mode = snapshotter.snapshot_mode
-    snapshotter.snapshot_dir = log_dir
-    snapshotter.snapshot_mode = args.snapshot_mode
-    snapshotter.snapshot_gap = args.snapshot_gap
+
     logger.push_prefix('[%s] ' % args.exp_name)
 
+    snapshotter = Snapshotter(
+        snapshot_dir=log_dir,
+        snapshot_mode=args.snapshot_mode,
+        snapshot_gap=args.snapshot_gap)
+
     if args.resume_from_dir is not None:
-        with LocalRunner() as runner:
-            runner.restore(args.resume_from_dir, from_epoch=args.resume_epoch)
+        with LocalRunner(
+                snapshotter=snapshotter,
+                sess=args.tf_session,
```

Comment:
hmm this might be a problem. people have real reasons to want to use their own TF session...